packages/core-data/README.md

```diff
@@ -285,14 +285,14 @@ Calling this may trigger an OPTIONS request to the REST API via the
 
 _Parameters_
 
--   _state_ `Object`: Data state.
+-   _state_ `State`: Data state.
 -   _kind_ `string`: Entity kind.
 -   _name_ `string`: Entity name.
--   _recordId_ `string`: Record's id.
+-   _recordId_ `RecordKey`: Record's id.
```

Comment:
I don't like how it says `RecordKey` here, I'd rather have it say `string | number`

Reply:
why not? because of the auto-generated documentation?

if that's the case I don't think we can care too much about this problem unless and until we radically change our docs. this is going to be a problem with more and more TypeScript code since this is a win for the types but masks the actual type in the docs. we need hyper-links in our docs to jump to and provide a tooltip description for the types like `RecordKey` here.

unless we get those hyperlinks/tooltips I don't know if there's a good workaround that we should chase.